pyqtgraph/graphicsItems/ImageItem.py

```diff
@@ -399,6 +421,8 @@ def setOpts(self, update: bool=True, **kwargs):
             self.menu = None
         if 'autoDownsample' in kwargs:
             self.setAutoDownsample(kwargs['autoDownsample'])
+        if 'nanPolicy' in kwargs:
+            self.nanPolicy = kwargs['nanPolicy']
```

Comment:
I think we want to call `self.setNanPolicy(kwargs['nanPolicy'])`.

`self.nanPolicy` should not be an attribute, but should be a method that gets the currently stored value.  I would suggest renaming to `self._nanPolicy` and have another method

```python
def nanPolicy(self):
    return self._nanPolicy
```

That will be more in line w/ Qt's getters and setters.